library/src/main/java/com/timehop/stickyheadersrecyclerview/StickyRecyclerHeadersDecoration.java

```diff
@@ -101,9 +103,20 @@ public void onDrawOver(Canvas canvas, RecyclerView parent, RecyclerView.State st
       boolean hasStickyHeader = mHeaderPositionCalculator.hasStickyHeader(itemView, mOrientationProvider.getOrientation(parent), position);
       if (hasStickyHeader || mHeaderPositionCalculator.hasNewHeader(position, mOrientationProvider.isReverseLayout(parent))) {
         View header = mHeaderProvider.getHeader(parent, position);
-        Rect headerOffset = mHeaderPositionCalculator.getHeaderBounds(parent, header, itemView, hasStickyHeader);
+        //re-use existing Rect, if any.
+        Rect headerOffset = mHeaderRects.get(position);
+        if (headerOffset == null) {
+          headerOffset = new Rect();
+          mHeaderRects.put(position, headerOffset);
+        }
+        mHeaderPositionCalculator.initHeaderBounds(headerOffset, parent, header, itemView, hasStickyHeader);
         mRenderer.drawHeader(parent, canvas, header, headerOffset);
-        mHeaderRects.put(position, headerOffset);
+      }
+    }
+    //remove unused rects, if any.
+    if (childCount < mHeaderRects.size()) {
```

Comment:
How about the situation when objects are removed from the list dynamically? In this case we might have inconsistent (yet probably harmless) state of `mHeaderRects` containing rects that correspond to non-existing positions in the list. This is a corner case though... It used to be solved automatically since `mHeaderRects` was cleared on every scroll move.